### PR TITLE
Adding Deployment Author file to cd-service

### PR DESCRIPTION
**Pull request by gsandok**
Creating file to hold Deployment Author and time of deployment within manifest repo
To be displayed in frontend in later PR

### CODE REVIEW

File: pkg/api/api.proto
Comment (hannesg) [inline pkg/api/api.proto:258]: Shouldn't these fields be in the Environment_Application?

File: services/cd-service/pkg/repository/transformer.go
Comment (hannesg) [inline services/cd-service/pkg/repository/transformer.go:874]: Please write the data as dedicated files (one per field) like it's done for locks here: https://github.com/freiheit-com/kuberpult/blob/300a1f54f5818b80a3bb338a5428612658fbcb0a/services/cd-service/pkg/repository/transformer.go#L607C11-L607C11

File: pkg/api/api.proto
Comment (hannesg) [inline pkg/api/api.proto:283]: Why do you have multiple environments in one again?

File: pkg/api/api.proto
Comment (sven-urbanski-freiheit-com) [inline pkg/api/api.proto:259]:
```suggestion
        int64 deployAtSecondsUtc = 2;
```
It should be clear that it's a timestamp, it's in UTC, and it should be an int of some form (maybe int32 is enough).

**Review by hannesg — COMMENTED** [review]
Hi. What exactly do you want to display in the frontend? Maybe it's easier to write a deployment log or something like this to the application version and then make a dedicated endpoint for that?

File: services/cd-service/pkg/repository/repository.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:904]: it's not yaml though, is it?
please remove the yaml suffix. Maybe replace with ".csv"

File: pkg/api/api.proto
Comment (hannesg) [reply pkg/api/api.proto:259]: int32 is never enough

File: services/cd-service/pkg/repository/repository.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:907]: ahhh, I'm just refactoring the defaultUser.
for now, please use the constant in auth.go.
But depending on who merges first, we need to adapt this.

File: services/cd-service/pkg/repository/repository.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:916]: here too

File: services/cd-service/pkg/repository/repository.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:914]: think this will fail if logs is an empty string

File: services/cd-service/pkg/repository/transformer.go
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go:874]: We will store multiple deployments. I think for this case it's more consistent to have one file with multiple entries.
Otherwise we can get into odd edgecases where there's 1 author and 2 timestamps.

(story: https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/-NQPLdkfXUNc4bE9KOiO/FILESYSTEM/FOLDER/ROOT/STORY/-NWiltvEUw2EQt4SdlvI?iteration=CURRENT_ITERATIONS&user=-NHUO4EmpZZtkcSSqr1i)

Alternative: We create a directory with an array (like "releases") and have single files in `deploymentMetadata/1/author` and `deploymentMetadata/1/deploymentTimestamp`

What do you think @hannesg ?

File: services/cd-service/pkg/repository/transformer_test.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer_test.go:401]: please also add a filed for `expectedDeploymentMetadataContent string` and test that the content is as expected.

File: services/cd-service/pkg/service/overview.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/overview.go:178]: since you have a return in the `then` part, 
please remove the `else` to avoid unnecessary nesting

File: pkg/api/api.proto
Comment (sven-urbanski-freiheit-com) [reply pkg/api/api.proto:258]: ahh yes, I think that was my suggestion. But now I see the redundancy. I also think it should go in the `Environment_Application`.

File: pkg/api/api.proto
Comment (sven-urbanski-freiheit-com) [inline pkg/api/api.proto:257]: to be consistent, we should write both the author name and author email.

File: services/cd-service/pkg/repository/repository.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:902]: should be:
`/environments/development/applications
/echo/deployment_author`
`/environments/development/applications
/echo/deployment_time_utc`

File: pkg/api/api.proto
Comment (sven-urbanski-freiheit-com) [inline pkg/api/api.proto:266]: it shouldn't be a string here, but an uint64

Comment (sven-urbanski-freiheit-com) [inline pkg/api/api.proto:266]:
```suggestion
      // we use a string here, because the UI cannot handle int64 as a type.
      // the string contains the unix timestamps in seconds (utc)
      string deployTime = 2;
```

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Comment by sven-urbanski-freiheit-com** [comment]
Before merging, please adapt the commit message, e.g.
> Store person who triggered deployment
>
> Stores users name and current date in the manifest repository.
> To be displayed in frontend in later PR.

File: services/cd-service/pkg/repository/repository.go
Comment (hannesg) [inline services/cd-service/pkg/repository/repository.go:899]: No email here?

**Review by hannesg — COMMENTED** [review]
Didn't you miss saving the email address? But the code looks nice!

File: services/cd-service/pkg/repository/transformer.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:873]: if you merge this after my change, you need to access the name like this:
```
	user, err := auth.ReadUserFromContext(ctx)
	if err != nil {
		return err
	}
       /// ... auth.Name
```

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go:873]: by "my change" i mean this: https://github.com/freiheit-com/kuberpult/pull/765

File: services/cd-service/pkg/repository/repository.go
Comment (gsandok) [reply services/cd-service/pkg/repository/repository.go:899]: Was not mentioned in the meeting we had on Thursday but can add now

**Review by hannesg — APPROVED** [review]

**Pull request merged**
